src/eos_gates.js

```diff
@@ -183,6 +183,27 @@ function installAppFromStore(remote, appId) {
     spawnProcess(['gnome-software', '--details=%s'.format(appStoreId)]);
 }
 
+function launchFlatpakApp(appId) {
+    try {
+        spawnProcess(['flatpak', 'run', appId]);
```

Comment:
That's what gnome-software uses too, and I'd be very surprised if it was broken. Sounds worth investigating at the very least.